### PR TITLE
redis: enable tests

### PR DESCRIPTION
**Pull request by risicle**
###### Description of changes

Enable tests on non-darwin platforms. Darwin is probably doable if we can get a pure `pgrep` for it. See #141157.

Disabling the flakiest and slowest tests seems to make this reliable enough for me.

There is also apparently a test mode which works against an "external" redis server and in theory we could use that in our nixos test, but I couldn't get it to work right :shrug: 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/nosql/redis/default.nix [hunk 1]
```diff
@@ -44,7 +45,26 @@ stdenv.mkDerivation rec {
...
+    runHook preCheck
+
+    # disable test "Connect multiple replicas at the same time": even
+    # upstream find this test too timing-sensitive
+    substituteInPlace tests/integration/replication.tcl \
+      --replace 'foreach mdl {no yes}' 'foreach mdl {}'
+
+    ./runtest \
+      --no-latency \
+      --timeout 2000 \
```
Comment (Mindavi) [inline pkgs/servers/nosql/redis/default.nix:61]: 2 seconds? Or 2000 seconds?

**Review by Mindavi — APPROVED** [review]
LGTM, didn't run them though

File: pkgs/servers/nosql/redis/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/servers/nosql/redis/default.nix:61]: Two thousand I'm afraid

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 164059` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>netbox</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>python310Packages.django-redis</li>
    <li>python310Packages.python-redis-lock</li>
    <li>python39Packages.django-redis</li>
    <li>python39Packages.python-redis-lock</li>
    <li>redis</li>
  </ul>
</details>

**Pull request merged**